src/earn/EarnEnterAmount.tsx

```diff
@@ -0,0 +1,626 @@
+                </>
+              </View>
+            </View>
+            <View style={styles.localAmountRow}>
+              <AmountInput
+                inputValue={token.priceUsd ? localAmountInput : '-'}
+                onInputChange={onLocalAmountInputChange}
+                inputRef={localAmountInputRef}
+                inputStyle={styles.localAmount}
+                placeholder={`${localCurrencySymbol}${new BigNumber(0).toFormat(2)}`}
+                testID="SendEnterAmount/LocalAmountInput"
+                editable={!!token.priceUsd}
+              />
+              {!token.balance.isZero() && (
+                <Touchable
+                  borderRadius={MAX_BORDER_RADIUS}
+                  onPress={onMaxAmountPress}
+                  style={styles.maxTouchable}
+                  testID="SendEnterAmount/Max"
+                >
+                  <Text style={styles.maxText}>{t('max')}</Text>
+                </Touchable>
+              )}
+            </View>
+          </View>
+        </View>
+        {showNotEnoughBalanceForGasWarning && (
+          <InLineNotification
+            variant={NotificationVariant.Warning}
+            title={t('sendEnterAmountScreen.notEnoughBalanceForGasWarning.title', {
+              feeTokenSymbol: prepareTransactionsResult.feeCurrencies[0].symbol,
+            })}
+            description={t('sendEnterAmountScreen.notEnoughBalanceForGasWarning.description', {
+              feeTokenSymbol: prepareTransactionsResult.feeCurrencies[0].symbol,
+            })}
+            style={styles.warning}
+            testID="SendEnterAmount/NotEnoughForGasWarning"
+          />
+        )}
+        {prepareTransactionError && (
+          <InLineNotification
+            variant={NotificationVariant.Error}
+            title={t('sendEnterAmountScreen.prepareTransactionError.title')}
+            description={t('sendEnterAmountScreen.prepareTransactionError.description')}
+            style={styles.warning}
+            testID="SendEnterAmount/PrepareTransactionError"
+          />
+        )}
+        <EarnProceed
+          tokenAmount={tokenAmount}
+          localAmount={localAmount}
+          token={token}
+          amountEnteredIn={enteredIn}
+          onPressProceed={onPressContinue}
+          onPressInfo={onPressInfo}
+          disabled={disabled}
+        />
+        <KeyboardSpacer />
+      </KeyboardAwareScrollView>
+      <InfoBottomSheet infoBottomSheetRef={infoBottomSheetRef} />
+      <EarnAddCryptoBottomSheet
+        forwardedRef={addCryptoBottomSheetRef}
+        token={token}
+        tokenAmount={tokenAmount ? tokenAmount.minus(token.balance) : new BigNumber(0)}
+      />
+      {prepareTransactionsResult?.type === 'possible' && (
+        <EarnDepositBottomSheet
+          forwardedRef={reviewBottomSheetRef}
+          preparedTransaction={prepareTransactionsResult}
+          amount={tokenAmount ? tokenAmount.toString() : '0'}
```

Comment:
tokenAmount should be defined with tx is possible right? You can just add this as another condition to the conditional rendering check above instead of the ternary here.